File: LeetCode/129-sum-root-to-leaf-numbers/solution.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def sumNumbers(self, root: TreeNode | None) -> int:
        return self.traverse(root, 0)
    
    def traverse(self, node: TreeNode | None, sum: int) -> int:
        sum = (sum*10)+node.val
        if (node.left is None and node.right is None):
            return sum
        left = 0 if node.left is None else self.traverse(node.left, sum)
        right = 0 if node.right is None else self.traverse(node.right, sum)
        return left + right

File: LeetCode/129-sum-root-to-leaf-numbers/Solution.java
class TreeNode {
	int val;
	TreeNode left;
	TreeNode right;
	TreeNode() {}
	TreeNode(int val) { this.val = val; }
	TreeNode(int val, TreeNode left, TreeNode right) {
		this.val = val;
		this.left = left;
		this.right = right;
	}
}

class Solution {
    public int sumNumbers(TreeNode root) {
        return sumNumbers(root, 0);
    }

    private int sumNumbers(TreeNode node, int runningSum) {
        if (node.left == null && node.right == null) return (runningSum*10)+node.val;
        int left = node.left == null ? 0 : sumNumbers(node.left, (runningSum*10)+node.val);
        int right = node.right == null ? 0 : sumNumbers(node.right, (runningSum*10)+node.val);
        return left+right;
    }
}